google/cloud/bigquery/job/query.py

```diff
@@ -1265,12 +1275,12 @@ def to_arrow(
     # changes to table.RowIterator.to_dataframe()
     def to_dataframe(
         self,
-        bqstorage_client=None,
-        dtypes=None,
-        progress_bar_type=None,
-        create_bqstorage_client=True,
-        date_as_object=True,
-    ):
+        bqstorage_client: bigquery_storage.BigQueryReadClient = None,
+        dtypes: Dict[str, Any] = None,
+        progress_bar_type: str = None,
+        create_bqstorage_client: bool = True,
+        date_as_object: bool = True,
+    ) -> Any:
```

Comment:
Why `Any` and not `"pandas.DataFrame"`?

Reply:
Perhaps because of [this](https://github.com/googleapis/python-bigquery/pull/445#discussion_r549215027)? I'll nevertheless check again if a quick workaround can be found.

Reply:
Good news - I figured out that if we install all dependencies in the nox session before running the type checks, A LOT of these import errors disappear, and we don't need to make workarounds such as this one or silencing the checker all around the code.